lib/relation-definition.js

```diff
@@ -1325,7 +1362,8 @@ BelongsTo.prototype.related = function (refresh, params) {
       if (query.where[pk] === undefined
         || query.where[pk] === null) {
         // Foreign key is undefined
-        return process.nextTick(cb);
+        process.nextTick(cb);
+        return cb.promise;
```

Comment:
AFAICT, `related()` does not support promise API, `cb.promise` is never set and this change can be reverted.